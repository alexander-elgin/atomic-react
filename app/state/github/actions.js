import {
  CHANGE_USERNAME,
  LOAD_REPOS,
  LOAD_REPOS_ERROR,
  LOAD_REPOS_SUCCESS,
} from './constants';

export function changeUsername(name) {
  return {
    type: CHANGE_USERNAME,
    name,
  };
}

export function loadRepositories() {
  return {
    type: LOAD_REPOS,
  };
}

export function setRepositories(repos, currentUsername) {
  return {
    type: LOAD_REPOS_SUCCESS,
    repos,
    currentUsername,
  };
}

export function setLoadRepositoriesError(error) {
  return {
    type: LOAD_REPOS_ERROR,
    error,
  };
}
