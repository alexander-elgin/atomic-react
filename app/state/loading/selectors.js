import { createSelector } from 'reselect';

const selectLoading = (state) => state.get('loading');

const makeSelectLoading = () => createSelector(
  selectLoading,
  (globalState) => globalState.get('loading')
);

export {
  selectLoading,
  makeSelectLoading,
};
