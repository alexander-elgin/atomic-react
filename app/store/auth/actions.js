import {
  SIGN_IN_SUCCESS,
  SIGN_OUT,
} from './constants';

export function setSignInData(user) {
  return {
    type: SIGN_IN_SUCCESS,
    user,
  };
}

export function signOut() {
  return {
    type: SIGN_OUT,
  };
}
