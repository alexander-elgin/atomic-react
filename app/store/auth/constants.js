export const SIGN_IN_SUCCESS = 'actions/auth/SIGN_IN_SUCCESS';
export const SIGN_OUT = 'actions/auth/SIGN_OUT';

export const ERRORS_PREFIX = 'boilerplate.shared.Auth.errors';

export const DUPLICATED_EMAIL = 'DUPLICATED_EMAIL';
export const EMPTY_PASSWORD = 'EMPTY_PASSWORD';
export const INCORRECT_CREDENTIALS = 'INCORRECT_CREDENTIALS';
export const INVALID_EMAIL = 'INVALID_EMAIL';
export const INVALID_NAME = 'INVALID_NAME';
export const INVALID_PASSWORD = 'INVALID_PASSWORD';
