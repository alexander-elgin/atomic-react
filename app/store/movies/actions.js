import {
  FETCH_MOVIES,
  FETCH_MOVIES_ERROR,
  FETCH_MOVIES_SUCCESS,
  RESET,
  SET_QUERY,
  SET_TOTAL_PAGES,
} from './constants';

export function fetchMovies() {
  return {
    type: FETCH_MOVIES,
  };
}

export function setError(error) {
  return {
    type: FETCH_MOVIES_ERROR,
    error,
  };
}

export function setMovies(movies) {
  return {
    type: FETCH_MOVIES_SUCCESS,
    movies,
  };
}

export function reset() {
  return {
    type: RESET,
  };
}

export function setQuery(query) {
  return {
    type: SET_QUERY,
    query,
  };
}

export function setTotalPages(totalPages) {
  return {
    type: SET_TOTAL_PAGES,
    totalPages,
  };
}
