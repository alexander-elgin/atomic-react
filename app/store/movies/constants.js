export const FETCH_GENRES = 'actions/movies/FETCH_GENRES';
export const FETCH_GENRES_SUCCESS = 'actions/movies/FETCH_GENRES_SUCCESS';
export const FETCH_MOVIES = 'actions/movies/FETCH_MOVIES';
export const FETCH_MOVIES_SUCCESS = 'actions/movies/FETCH_MOVIES_SUCCESS';
export const RESET = 'actions/movies/RESET';
export const SET_ERROR = 'actions/movies/SET_ERROR';
export const SET_QUERY = 'actions/movies/SET_QUERY';
export const SET_TOTAL_PAGES = 'actions/movies/SET_TOTAL_PAGES';
