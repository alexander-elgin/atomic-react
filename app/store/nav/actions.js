import {
  SET_PAGE_TITLE,
} from './constants';

export function setPageTitle(payload) {
  return {
    type: SET_PAGE_TITLE,
    payload,
  };
}
