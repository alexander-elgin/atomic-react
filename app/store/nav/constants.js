export const SET_PAGE_TITLE = 'actions/nav/SET_PAGE_TITLE';
