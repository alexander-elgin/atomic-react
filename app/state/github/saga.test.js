import { put, takeLatest } from 'redux-saga/effects';

import { LOAD_REPOS } from './constants';
import { setRepositories, setLoadRepositoriesError } from './actions';
import { resetLoading } from '../loading/actions';

import githubData, { getRepositories } from './saga';

const username = 'alexander-elgin';

/* eslint-disable redux-saga/yield-effects */
describe('github Saga', () => {
  let getRepositoriesGenerator;

  beforeEach(() => {
    getRepositoriesGenerator = getRepositories();

    const selectDescriptor = getRepositoriesGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = getRepositoriesGenerator.next(username).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('dispatches the setRepositories action if it fetches the repositories data successfully', () => {
    const repositories = [
      {
        name: 'First repo',
      },
      {
        name: 'Second repo',
      },
    ];
    const putDescriptor = getRepositoriesGenerator.next(repositories).value;
    expect(putDescriptor).toEqual(put(setRepositories(repositories, username)));
  });

  it('dispatches the setLoadRepositoriesError action if the request fails', () => {
    const response = new Error('Some error');
    const putDescriptor = getRepositoriesGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(setLoadRepositoriesError(response)));
  });

  afterEach(() => {
    const putDescriptor = getRepositoriesGenerator.next().value;
    expect(putDescriptor).toEqual(put(resetLoading()));
  });
});

describe('githubDataSaga Saga', () => {
  const githubDataSaga = githubData();

  it('should start task to watch for LOAD_REPOS action', () => {
    const takeLatestDescriptor = githubDataSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(LOAD_REPOS, getRepositories));
  });
});
