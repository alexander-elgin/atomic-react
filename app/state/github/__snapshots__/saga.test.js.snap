// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`github Saga dispatches the setLoadRepositoriesError action if the request fails 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`github Saga dispatches the setLoadRepositoriesError action if the request fails 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://api.github.com/users/alexander-elgin/repos?type=all&sort=updated",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`github Saga dispatches the setRepositories action if it fetches the repositories data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`github Saga dispatches the setRepositories action if it fetches the repositories data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://api.github.com/users/alexander-elgin/repos?type=all&sort=updated",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
